noir/noir-repo/tooling/debugger/src/repl.rs

```diff
@@ -319,12 +319,12 @@ impl<'a, B: BlackBoxFunctionSolver> ReplDebugger<'a, B> {
             return;
         };
 
-        for (index, value) in memory.iter().enumerate() {
-            println!("{index} = {}", value.to_field());
+        for (index, value) in memory.iter().enumerate().filter(|(_, value)| value.bit_size > 0) {
```

Comment:
This is interesting. Where do we create zero bitsize memory values?

Reply:
Wondering if this is necessary or not.

Reply:
All memory default values (values never written to) are 0 field with a bit size of 0! With this change we avoid showing empty memory regions. Previously printing memory wasn't very useful since you got <reserved><populated stack><hundreds of zeroes of unpopulated stack><heap>

Reply:
Gotcha, thanks.

Reply:
To further clarify, memory is expanded as written. Since the first ~1000 items are reserved for stack, whenever the program writes to heap, memory is expanded up to the first heap index with default values, as can be seen here: https://github.com/AztecProtocol/aztec-packages/blob/arv/brillig_typed_memory/noir/noir-repo/acvm-repo/brillig_vm/src/memory.rs#L194